### PR TITLE
More unified build switches

### PR DESCRIPTION
**Pull request by mmitche**
- Add DotNetBuildOrchestrator usage
- Add ExcludeFromDotNetBuild
- Add ExcludeFromSourceOnlyBuild
- Add test skipping switch. On by default in source-only modes. Off by default in normal .net build mode.
- Add uses of DotNetBuildSourceOnly
- Add uses of ExcludeFromSourceOnlyBuild

### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

**Comment by mmitche** [comment]
Needs more validation before this goes in.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Localization.targets [hunk 1]
```diff
@@ -22,18 +22,20 @@
...
   <ItemGroup>
-    <PackageReference Include="Microsoft.DotNet.XliffTasks" Version="$(MicrosoftDotNetXliffTasksVersion)" PrivateAssets="all" IsImplicitlyDefined="true" Condition="'$(UsingToolXliff)' == 'true' and '$(IsShippingAssembly)' == 'true'" />
+    <PackageReference Include="Microsoft.DotNet.XliffTasks" Version="$(MicrosoftDotNetXliffTasksVersion)" PrivateAssets="all" IsImplicitlyDefined="true"
+      Condition="'$(UsingToolXliff)' == 'true' and '$(IsShippingAssembly)' == 'true'" />
   </ItemGroup>
 
 
   <!-- TemplateLocalizer for localizing 'dotnet new' templates -->
-  <PropertyGroup Condition="'$(UsingToolTemplateLocalizer)' == 'true' AND '$(DotNetBuildFromSource)' != 'true'">
+  <PropertyGroup Condition="'$(UsingToolTemplateLocalizer)' == 'true' AND '$(DotNetBuildFromSource)' != 'true' AND '$(DotNetBuildSourceOnly)' != 'true'">
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Localization.targets:31]: nit: let's make this consistent with most other msbuild code:

```suggestion
  <PropertyGroup Condition="'$(UsingToolTemplateLocalizer)' == 'true' and '$(DotNetBuildFromSource)' != 'true' and '$(DotNetBuildSourceOnly)' != 'true'">
```

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Localization.targets [hunk 2]
```diff
@@ -22,18 +22,20 @@
...
     <!-- Run localizer when building on dev machine. -->
     <LocalizeTemplates Condition="'$(ContinuousIntegrationBuild)' != 'true'">true</LocalizeTemplates>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.TemplateEngine.Authoring.Tasks" Version="$(MicrosoftTemplateEngineAuthoringTasksVersion)" PrivateAssets="all" IsImplicitlyDefined="true" Condition="'$(UsingToolTemplateLocalizer)' == 'true' AND '$(DotNetBuildFromSource)' != 'true'" />
+    <PackageReference Include="Microsoft.TemplateEngine.Authoring.Tasks" Version="$(MicrosoftTemplateEngineAuthoringTasksVersion)" PrivateAssets="all"
+      IsImplicitlyDefined="true" Condition="'$(UsingToolTemplateLocalizer)' == 'true' AND '$(DotNetBuildFromSource)' != 'true' AND '$(DotNetBuildSourceOnly)' != 'true'" />
```
Comment (ViktorHofer) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Localization.targets:38]:
```suggestion
      IsImplicitlyDefined="true" Condition="'$(UsingToolTemplateLocalizer)' == 'true' and '$(DotNetBuildFromSource)' != 'true' and '$(DotNetBuildSourceOnly)' != 'true'" />
```

**Review by ViktorHofer — APPROVED** [review]

**Review by ViktorHofer — APPROVED** [review]

**Comment by mmitche** [comment]
Feeling pretty good about these right now. After this and #14384 goes in and it flows into the VMR, I think we can actually begin to remove DotNetBuildVertical.

**Review by ViktorHofer — APPROVED** [review]

**Pull request merged**